### PR TITLE
revert supports[:manage_home] behavior

### PR DESCRIPTION
**Pull request by lamont-granquist**
even though i violently disagree that this is correct behavior and we're
just going to break everyone in one massive go when we hit Chef 13.

Signed-off-by: Lamont Granquist <lamont@scriptkiddie.org>

### CODE REVIEW

**Comment by lamont-granquist** [comment]
@thommay probably up to you as to if this is a regression or not.

should probably get feedback from an affected user that this reverts to the old behavior of supports[:manage_home]

**Comment by coderanger** [comment]
👍 from me, but I also defer to @thommay's wisdom.

**Comment by bradenwright** [comment]
@lamont-granquist lol at the "violently" disagree, I may steal that

**Comment by lamont-granquist** [comment]
pretty certain i stole it from someone on Usenet around 1992 (or some dragcit BBS or something back in the dawn of time)

File: lib/chef/resource/user/linux_user.rb [hunk 1]
```diff
@@ -29,21 +29,24 @@ class LinuxUser < Chef::Resource::User
...
 
         def supports(args = {})
-          Chef.log_deprecation "setting supports on the linux_user resource is deprecated"
-          # setting is deliberately disabled
-          super({})
+          if args.key?(:manage_home)
+            Chef.log_deprecation "supports { manage_home: #{args[:manage_home]} } on the user resource is deprecated and will be removed in Chef 13, set manage_home: #{args[:manage_home]} instead"
+          end
+          if args.key?(:non_unique)
+            Chef.log_deprecation "supports { manage_home: #{args[:non_unique]} } on the user resource is deprecated and will be removed in Chef 13, set manage_home: #{args[:non_unique]} instead"
```
Comment (gregkare) [inline lib/chef/resource/user/linux_user.rb]: This should be `non_unique` everywhere on this line, not `manage_home`

Comment (lamont-granquist) [inline lib/chef/resource/user/linux_user.rb]: fixed

**Comment by mwrock** [comment]
:+1:

**Pull request merged**